host/volume/backend.go

```diff
@@ -17,6 +18,12 @@ type Provider interface {
 
 	NewVolume() (Volume, error)
 	DestroyVolume(Volume) error
+	CreateSnapshot(Volume) (Volume, error)
+	ForkVolume(Volume) (Volume, error)
+
+	ReportHaves(Volume) ([]json.RawMessage, error) // Report known data addresses; this can be given to `SendSnapshot` to attempt deduplicated/incrememntal transport.
```

Comment:
I borrowed the term `haves` from [git's transport packing spec](https://github.com/git/git/blob/18d0fec24027ac226dc2c4df2b955eef2a16462a/Documentation/technical/pack-protocol.txt#L283).  I'd be open to different names, but specifically opposed to calling it `ListSnapshots` because that's an implementation detail of ZFS (and a bad one at that; insert entire rant here about how ZFS fails miserably to make a distinction between the object layer and the file tree layer, which goes a long way to explaining why its dedup is such here-be-dragons territory to begin with).

I can't currently think of any providers or protocols where much beyond a string objectID is needed here, so I guess I tossed a json object in mostly to leave room for future upgrades.  I could see arguments for either cutting to `[]string` or doing one big `json.RawMessage`.  Thoughts?  At the end of the day, the only constraint is that it's going in an API post body shortly.